### PR TITLE
Add theta-formulation support

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR adds theta-liquid-ice formulation support.

Co-authored with @OsKnoth, @slifer50, @smarras79, @akshaysridhar, @bischtob. 

### Description

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/Atmos/Model/tendencies_energy.jl [hunk 1]
```diff
@@ -40,7 +52,7 @@ struct DiffEnthalpyFlux{PV <: Energy} <: TendencyDef{Flux{SecondOrder}, PV} end
 function flux(::DiffEnthalpyFlux{Energy}, atmos, args)
     @unpack state, diffusive = args
     @unpack D_t = args.precomputed.turbulence
-    d_h_tot = -D_t .* diffusive.∇h_tot
+    d_h_tot = -D_t .* diffusive.energy.∇h_tot
```
Comment (tapios) [inline src/Atmos/Model/tendencies_energy.jl:55]: I didn't see where `energy.∇h_tot` is calculated. For `theta_li`, it's important that this is `theta_li` itself, not some other (enthalpy) variable. Is it?

Comment (charleskawczynski) [reply src/Atmos/Model/tendencies_energy.jl:55]: It's computed [here](https://github.com/CliMA/ClimateMachine.jl/pull/1935/files#diff-4032d710c4d46f450b770888b8ae491bd3f2134e6b1299a21170040198b070e5R52), for the energy formulation only.

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Comment by charleskawczynski** [comment]
bors try

**Review by bischtob — APPROVED** [review]
LGTM

File: src/Diagnostics/atmos_les_default.jl [hunk 2]
```diff
@@ -152,6 +152,7 @@ function atmos_les_default_simple_sums!(
     MH,
     sums,
 )
+    atmos.energy isa EnergyModel || error("EnergyModel only supported")
```
Comment (kpamnany) [inline src/Diagnostics/atmos_les_default.jl:155]: This should go into `atmos_les_default_collect`, not here.

File: experiments/AtmosLES/rising_bubble_theta_formulation.jl [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+# pre-defined functions from other packages. Julia allows us to use existing
+# modules (variable workspaces), or write our own to do so.  Complete
+# documentation for the Julia module system can be found
+# [here](https://docs.julialang.org/en/v1/manual/modules/#).
+
+# We need to use the `ClimateMachine` module! This imports all functions
+# specific to atmospheric and ocean flow modeling.
+
+using ClimateMachine
+ClimateMachine.init(; parse_clargs = true)
```
Comment (charleskawczynski) [inline experiments/AtmosLES/rising_bubble_theta_formulation.jl:68]: @kpamnany should this be changed? Or is it okay?

File: src/Atmos/Model/AtmosModel.jl [hunk 4]
```diff
@@ -307,7 +332,7 @@ function vars_state(m::AtmosModel, st::Prognostic, FT)
         # start of inclusion in `AtmosLinearModel`
         ρ::FT
         ρu::SVector{3, FT}
-        ρe::FT
+        energy::vars_state(m.energy, st, FT) # TODO: adjust linearmodel
```
Comment (charleskawczynski) [inline src/Atmos/Model/AtmosModel.jl:335]: I believe @slifer50 will address this in a subsequent PR.

File: src/Atmos/Model/energy.jl [hunk 5]
```diff
@@ -0,0 +1,97 @@
+export AbstractEnergyModel, EnergyModel, θModel
+#### Energy component in atmosphere model
+abstract type AbstractEnergyModel <: BalanceLaw end
```
Comment (kpamnany) [inline src/Atmos/Model/energy.jl:3]: Does this have to be a `BalanceLaw`?

File: src/Atmos/Model/atmos_tendencies.jl [hunk 6]
```diff
@@ -33,8 +33,21 @@ eq_tends(pv::PV, m::AtmosModel, ::Flux{FirstOrder}) where {PV <: Momentum} =
...
-    (Advect{PV}(), Pressure{PV}(), eq_tends(pv, m.radiation, tt)...)
+eq_tends(pv::PV, m::EnergyModel, tt::Flux{FirstOrder}) where {PV <: Energy} =
+    (Advect{PV}(), Pressure{PV}())
+
+eq_tends(pv::PV, m::θModel, tt::Flux{FirstOrder}) where {PV <: ρθ_liq_ice} =
+    (Advect{PV}(),)
+
+# TODO: make radiation aware of which energy formulation is used:
+# eq_tends(pv::PV, m::AtmosModel, tt::Flux{FirstOrder}) where {PV <: AbstractEnergy} =
+#     (eq_tends(pv, m.energy, tt)..., eq_tends(pv, m.energy, m.radiation, tt)...)
```
Comment (charleskawczynski) [inline src/Atmos/Model/atmos_tendencies.jl:44]: I think we can leave the existing hard-coded version for now and adjust this, if needed in a subsequent PR.

File: experiments/AtmosLES/rising_bubble_theta_formulation.jl (hunk 3 above)
Comment (kpamnany) [reply experiments/AtmosLES/rising_bubble_theta_formulation.jl:68]: This is fine I think. Only stuff in the `test/runtests.jl` hierarchy shouldn't parse command line arguments.

File: src/Atmos/Model/energy.jl (hunk 5 above)
Comment (charleskawczynski) [reply src/Atmos/Model/energy.jl:3]: No, I just added so that we don't need to define `vars_state` for all `AbstractStateType`'s

Comment (charleskawczynski) [reply src/Atmos/Model/energy.jl]: I removed the BL subtype

**Pull request merged**